components/prism-hoon.js

```diff
@@ -0,0 +1,30 @@
+Prism.languages.hoon = {
+	constant: /(?:%\.n)|(?:%\.y)|(?:%[\w-]+\w?)/,
+	comment: {
+		pattern: /::.*/,
+		lookahead: true,
+		greedy: true
+	},
+	'function-name': /(?:\+[-+] {2})?(?:[a-z]([a-z0-9-]*[a-z0-9])?)/,
```

Comment:
1. `function-name` is a non-standard name not supported by any themes. Please use `function` instead.
2. Shouldn't the `(?:\+[-+] {2})?` be a lookbehind?

Reply:
I think I used Rust's definition as reference there. I can change it.

Wouldn't a lookbehind be looking for `  ++`? You need two spaces between any rune any its arguments, so we want `++  `.

Reply:
The inline code on GH shortened the starting two spaces in that first inline code block.

Reply:
I meant the lookbehind like this:

```
'function': {
  pattern: /(\+[-+] {2})?(?:[a-z]([a-z0-9-]*[a-z0-9])?)/,
  lookbehind: true
}
```

The `\+[-+] {2}` is not part of the function name itself, or is it?

Reply:
You can define an arm by writing `++  name` in a core; and then refer to it (if it's in the environment) as `name`. Now maybe you consider `++` as a rune (or "keyword" here), but thus far in practice we group the arm name alongside its runic constructor in our pre-existing syntax highlighters, so it's all grouped here.

Reply:
Ok. In that case, a lookbehind isn't necessary.